src/coreclr/vm/interpexec.cpp

```diff
@@ -291,6 +293,67 @@ template <typename TResult, typename TSource> void ConvOvfHelper(int8_t *stack,
     }
 }
 
+#define DO_GENERIC_LOOKUP(genericVar, lookupIndex) \
+                    InterpGenericLookup *pLookup = (InterpGenericLookup*)pMethod->pDataItems[ip[lookupIndex]]; \
+                    MethodTable *pMT = nullptr;                                                             \
+                    MethodDesc* pMD = nullptr;                                                              \
+                    uint8_t* lookup;                                                                        \
+                    if (pLookup->lookupType == InterpGenericLookupType::This)                               \
+                    {                                                                                       \
+                        OBJECTREF thisPtr = LOCAL_VAR(genericVar, OBJECTREF);                               \
+                        NULL_CHECK(thisPtr);                                                                \
+                        pMT = thisPtr->GetMethodTable();                                                    \
+                        lookup = (uint8_t*)pMT;                                                             \
+                    }                                                                                       \
+                    else if (pLookup->lookupType == InterpGenericLookupType::Class)                         \
+                    {                                                                                       \
+                        pMT = LOCAL_VAR(genericVar, MethodTable*);                                          \
+                        lookup = (uint8_t*)pMT;                                                             \
+                    }                                                                                       \
+                    else                                                                                    \
+                    {                                                                                       \
+                        assert(pLookup->lookupType == InterpGenericLookupType::Method);                     \
+                        pMD = LOCAL_VAR(genericVar, MethodDesc*);                                           \
+                        lookup = (uint8_t*)pMD;                                                             \
+                    }                                                                                       \
+                    void* result = 0;                                                                       \
```

Comment:
[nitpick] It'd be nice if this name made it a little clearer that it's from the macro, or if it was defined at the top of the macro instead of hiding in the middle